.github/workflows/release.yaml

```diff
@@ -0,0 +1,45 @@
+name: Release
+on:
+  push:
+    tags:
+      - 'v*.*.*'
+jobs:
+  build-and-push-image:
+    env:
+      IMAGE_NAME: autoinstrumentation-go
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
```

Comment:
Can we pin all GitHub Actions' verions?\
E.g. `actions/checkout@v3.3.0`.
This would follow the same pattern as in https://github.com/open-telemetry/opentelemetry-go-instrumentation/blob/main/.github/workflows/links.yml

Dependabot is making a PRs to bump and we would mitigate the possible risk if a new version of a GitHub Action would contain a bug.